### PR TITLE
Skylink fix

### PR DESCRIPTION
**Pull request by arvvoid**
## Please check that your pull request pass this checks:

-  [x] My changes respect the [Kodi add-on development rules](https://kodi.wiki/view/Add-on_rules)
-  [x] My code is both Python 2 and 3 compatible
-  [x] My code follows the code style of this project
-  [x] I made sure there wasn't another Pull Request opened for the same update/change
-  [x] I have successfully tested my changes locally

## Types of changes

-  [X] Bug fix (non-breaking change which fixes an issue)
-  [ ] New feature (non-breaking change which adds functionality)
-  [ ] Breaking change (fix or feature that would cause existing functionality to change)
-  [ ] Adding a new region/hbo go version

## Description:

This should fix Skylink CZ and Slovakia operator based on @ferdabasek code, the code is ready for merging after testing.

Linked to ISSUE #5

### CODE REVIEW

**Comment by arvvoid** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/plugin.video.hbogoeu/plugin.video.hbogoeu/pullRequest?prid=4582076&bid=15390820)

File: hbogolib/handlereu.py [hunk 1]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
...
 
             self.log("LOGIN FORM PAYLOAD: " + str(payload))
 
+            if self.op_id == HbogoConstants.SkylinkID:
+               thumb = re.compile('<input type="hidden" name="__VIEWSTATE" id="__VIEWSTATE" value="(.+?)" />').findall(r.text)
+               thumb2 = re.compile('<input type="hidden" name="__VIEWSTATEGENERATOR" id="__VIEWSTATEGENERATOR" value="(.+?)" />').findall(r.text)
+               thumb3 = re.compile('<input type="hidden" name="__EVENTVALIDATION" id="__EVENTVALIDATION" value="(.+?)" />').findall(r.text)
+               payload['__EVENTTARGET'] = 'btnSubmit'
+               payload['__VIEWSTATE'] = thumb[0]
+               payload['__VIEWSTATEGENERATOR'] = thumb2[0]
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Bad indentation. Found 15 spaces, expected 16](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 2]
```diff
@@ -19,6 +19,8 @@
 
 import requests
 
+import re
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [standard import "import re" should be placed before "import requests"](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 3]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
...
             self.log("LOGIN FORM PAYLOAD: " + str(payload))
 
+            if self.op_id == HbogoConstants.SkylinkID:
+               thumb = re.compile('<input type="hidden" name="__VIEWSTATE" id="__VIEWSTATE" value="(.+?)" />').findall(r.text)
+               thumb2 = re.compile('<input type="hidden" name="__VIEWSTATEGENERATOR" id="__VIEWSTATEGENERATOR" value="(.+?)" />').findall(r.text)
+               thumb3 = re.compile('<input type="hidden" name="__EVENTVALIDATION" id="__EVENTVALIDATION" value="(.+?)" />').findall(r.text)
+               payload['__EVENTTARGET'] = 'btnSubmit'
+               payload['__VIEWSTATE'] = thumb[0]
+               payload['__VIEWSTATEGENERATOR'] = thumb2[0]
+               payload['__EVENTVALIDATION'] = thumb3[0]
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Bad indentation. Found 15 spaces, expected 16](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 4]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
...
+                   self.REDIRECT_URL.split('?')[0],
+                   params=payload
+                )
 
             payload = HbogoConstants.eu_redirect_login[self.op_id][3]
 
             self.log("LOGIN FORM PAYLOAD: " + str(payload))
 
+            if self.op_id == HbogoConstants.SkylinkID:
+               thumb = re.compile('<input type="hidden" name="__VIEWSTATE" id="__VIEWSTATE" value="(.+?)" />').findall(r.text)
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Bad indentation. Found 15 spaces, expected 16](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 5]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
...
+               thumb2 = re.compile('<input type="hidden" name="__VIEWSTATEGENERATOR" id="__VIEWSTATEGENERATOR" value="(.+?)" />').findall(r.text)
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Bad indentation. Found 15 spaces, expected 16](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 6]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
...
+               thumb3 = re.compile('<input type="hidden" name="__EVENTVALIDATION" id="__EVENTVALIDATION" value="(.+?)" />').findall(r.text)
+               payload['__EVENTTARGET'] = 'btnSubmit'
+               payload['__VIEWSTATE'] = thumb[0]
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Bad indentation. Found 15 spaces, expected 16](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 7]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
 
             self.log("GET CP SESSION: " + self.REDIRECT_URL.split('?')[0])
 
-            cp_session.get(
-                self.REDIRECT_URL.split('?')[0],
-                params=payload
-            )
+            r = cp_session.get(
+                   self.REDIRECT_URL.split('?')[0],
+                   params=payload
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Wrong hanging indentation (remove 3 spaces).](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 8]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
...
 
             payload = HbogoConstants.eu_redirect_login[self.op_id][3]
 
             self.log("LOGIN FORM PAYLOAD: " + str(payload))
 
+            if self.op_id == HbogoConstants.SkylinkID:
+               thumb = re.compile('<input type="hidden" name="__VIEWSTATE" id="__VIEWSTATE" value="(.+?)" />').findall(r.text)
+               thumb2 = re.compile('<input type="hidden" name="__VIEWSTATEGENERATOR" id="__VIEWSTATEGENERATOR" value="(.+?)" />').findall(r.text)
+               thumb3 = re.compile('<input type="hidden" name="__EVENTVALIDATION" id="__EVENTVALIDATION" value="(.+?)" />').findall(r.text)
+               payload['__EVENTTARGET'] = 'btnSubmit'
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Bad indentation. Found 15 spaces, expected 16](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/constants.py [hunk 9]
```diff
@@ -98,14 +98,15 @@ class HbogoConstants(object):
 
     # 0 - operator website login form url, 1 - username field name, 2 - password field name, 3 form payload
     eu_redirect_login = {
-        'c55e69f0-2471-46a9-a8b7-24dac54e6eb9': ['https://hbogo.skylink.cz/goauthenticate.aspx?client_id=HBO&redirect_uri=https%3a%2f%2fczapi.hbogo.eu%2foauthskylink%2frequest2.aspx&state=8wp6mMCCUjVJZEQdcb0BQA&scope=HBO&response_type=code', 'txtLogin', 'txtPassword', {"txtLogin": None, "txtPassword": None, "__LASTFOCUS": "", "__EVENTTARGET": "", "__EVENTARGUMENT": "", "__VIEWSTATE": ""}],  # Czech Republic: Skylink + Slovakia: Skylink
+        'c55e69f0-2471-46a9-a8b7-24dac54e6eb9': ['https://hbogo.skylink.cz/goauthenticate.aspx?client_id=HBO&redirect_uri=https%3a%2f%2fczapi.hbogo.eu%2foauthskylink%2frequest2.aspx&state=5zveHRYBaocYXvjTxHozRg&scope=HBO&response_type=code', 'txtLogin', 'txtPassword', {"__LASTFOCUS": None, "__EVENTTARGET": None, "__EVENTARGUMENT": None, "__VIEWSTATE": None, "__VIEWSTATEGENERATOR": None, "txtLogin": None, "txtPassword": None}],  # Czech Republic: Skylink + Slovakia: Skylink
```
Comment (arvvoid) [inline hbogolib/constants.py]: Codacy Issue found: [line too long (477 > 159 characters)](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 10]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
...
+                )
 
             payload = HbogoConstants.eu_redirect_login[self.op_id][3]
 
             self.log("LOGIN FORM PAYLOAD: " + str(payload))
 
+            if self.op_id == HbogoConstants.SkylinkID:
+               thumb = re.compile('<input type="hidden" name="__VIEWSTATE" id="__VIEWSTATE" value="(.+?)" />').findall(r.text)
+               thumb2 = re.compile('<input type="hidden" name="__VIEWSTATEGENERATOR" id="__VIEWSTATEGENERATOR" value="(.+?)" />').findall(r.text)
+               thumb3 = re.compile('<input type="hidden" name="__EVENTVALIDATION" id="__EVENTVALIDATION" value="(.+?)" />').findall(r.text)
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Bad indentation. Found 15 spaces, expected 16](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 11]
```diff
@@ -443,15 +445,24 @@ def OAuthLogin(self, username, password):
 
             self.log("GET CP SESSION: " + self.REDIRECT_URL.split('?')[0])
 
-            cp_session.get(
-                self.REDIRECT_URL.split('?')[0],
-                params=payload
-            )
+            r = cp_session.get(
+                   self.REDIRECT_URL.split('?')[0],
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Wrong hanging indentation (remove 3 spaces).](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

**Comment by arvvoid** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/plugin.video.hbogoeu/plugin.video.hbogoeu/pullRequest?prid=4582076&bid=15390820)

File: hbogolib/handlereu.py [hunk 12]
```diff
@@ -443,15 +443,24 @@ def OAuthLogin(self, username, password):
...
+                   params=payload
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Wrong hanging indentation (remove 3 spaces).](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 13]
```diff
@@ -443,15 +443,24 @@ def OAuthLogin(self, username, password):
 
             self.log("GET CP SESSION: " + self.REDIRECT_URL.split('?')[0])
 
-            cp_session.get(
-                self.REDIRECT_URL.split('?')[0],
-                params=payload
-            )
+            r = cp_session.get(
+                   self.REDIRECT_URL.split('?')[0],
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [Wrong hanging indentation (remove 3 spaces).](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/constants.py [hunk 14]
```diff
@@ -96,9 +96,11 @@ class HbogoConstants(object):
 
     }
 
+    SkylinkID = "c55e69f0-2471-46a9-a8b7-24dac54e6eb9"  # Skylink Operator ID, Skylink require special steps in login
+
     # 0 - operator website login form url, 1 - username field name, 2 - password field name, 3 form payload
     eu_redirect_login = {
-        'c55e69f0-2471-46a9-a8b7-24dac54e6eb9': ['https://hbogo.skylink.cz/goauthenticate.aspx?client_id=HBO&redirect_uri=https%3a%2f%2fczapi.hbogo.eu%2foauthskylink%2frequest2.aspx&state=8wp6mMCCUjVJZEQdcb0BQA&scope=HBO&response_type=code', 'txtLogin', 'txtPassword', {"txtLogin": None, "txtPassword": None, "__LASTFOCUS": "", "__EVENTTARGET": "", "__EVENTARGUMENT": "", "__VIEWSTATE": ""}],  # Czech Republic: Skylink + Slovakia: Skylink
+        'c55e69f0-2471-46a9-a8b7-24dac54e6eb9': ['https://hbogo.skylink.cz/goauthenticate.aspx?client_id=HBO&redirect_uri=https%3a%2f%2fczapi.hbogo.eu%2foauthskylink%2frequest2.aspx&state=5zveHRYBaocYXvjTxHozRg&scope=HBO&response_type=code', 'txtLogin', 'txtPassword', {"__LASTFOCUS": None, "__EVENTTARGET": "btnSubmit", "__EVENTARGUMENT": None, "__VIEWSTATE": None, "__VIEWSTATEGENERATOR": None, "txtLogin": None, "txtPassword": None}],  # Czech Republic: Skylink + Slovakia: Skylink
```
Comment (arvvoid) [inline hbogolib/constants.py]: Codacy Issue found: [line too long (484 > 159 characters)](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

**Comment by arvvoid** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/plugin.video.hbogoeu/plugin.video.hbogoeu/pullRequest?prid=4582076&bid=15390820)

File: hbogolib/handlereu.py [hunk 15]
```diff
@@ -443,15 +443,18 @@ def OAuthLogin(self, username, password):
...
 
             payload = HbogoConstants.eu_redirect_login[self.op_id][3]
 
             self.log("LOGIN FORM PAYLOAD: " + str(payload))
 
+            if self.op_id == HbogoConstants.SkylinkID:  # Perform special steps for Skylink
+                import re
+                payload['__VIEWSTATE'] = re.compile('<input type="hidden" name="__VIEWSTATE" id="__VIEWSTATE" value="(.+?)" />').findall(r.text)[0]
+                payload['__VIEWSTATEGENERATOR'] = re.compile('<input type="hidden" name="__VIEWSTATEGENERATOR" id="__VIEWSTATEGENERATOR" value="(.+?)" />').findall(r.text)[0]
+                payload['__EVENTVALIDATION'] = re.compile('<input type="hidden" name="__EVENTVALIDATION" id="__EVENTVALIDATION" value="(.+?)" />').findall(r.text)[0]
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [line too long (165 > 159 characters)](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

File: hbogolib/handlereu.py [hunk 16]
```diff
@@ -443,15 +443,18 @@ def OAuthLogin(self, username, password):
...
+            r = cp_session.get(self.REDIRECT_URL.split('?')[0], params=payload)
 
             payload = HbogoConstants.eu_redirect_login[self.op_id][3]
 
             self.log("LOGIN FORM PAYLOAD: " + str(payload))
 
+            if self.op_id == HbogoConstants.SkylinkID:  # Perform special steps for Skylink
+                import re
+                payload['__VIEWSTATE'] = re.compile('<input type="hidden" name="__VIEWSTATE" id="__VIEWSTATE" value="(.+?)" />').findall(r.text)[0]
+                payload['__VIEWSTATEGENERATOR'] = re.compile('<input type="hidden" name="__VIEWSTATEGENERATOR" id="__VIEWSTATEGENERATOR" value="(.+?)" />').findall(r.text)[0]
```
Comment (arvvoid) [inline hbogolib/handlereu.py]: Codacy Issue found: [line too long (174 > 159 characters)](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4582076)

**Review by durip — APPROVED** [review]

File: changelog.md [hunk 17]
```diff
@@ -1,3 +1,7 @@
+v.2.0.23-beta39
+- THIS IS A WORK IN PROGRESS
+- Fix Skylink CZ and PL (contribution ferdabasek)
```
Comment (durip) [inline changelog.md]: Not PL but SK

File: addon.xml [hunk 18]
```diff
@@ -71,6 +71,10 @@ Minden HBO Go tartalom © 2019 Copyright © 2017 Home Box Office, Inc. Minden jo
     <source>https://github.com/arvvoid/plugin.video.hbogoeu</source>
     <website>https://arvvoid.github.io/plugin.video.hbogoeu</website>
     <news>
+v.2.0.23-beta39
+- THIS IS A WORK IN PROGRESS
+- Fix Skylink CZ and PL (contribution ferdabasek)
```
Comment (durip) [inline addon.xml]: Not PL but SK

File: changelog.md (hunk 17 above)
Comment (arvvoid) [reply changelog.md]: Yes tnx will correct that

File: addon.xml (hunk 18 above)
Comment (arvvoid) [reply addon.xml]: Yes tnx will correct that

**Comment by arvvoid** [comment]
Please test and report results: https://github.com/arvvoid/plugin.video.hbogoeu/archive/Skylink_fix.zip

https://github.com/arvvoid/plugin.video.hbogoeu/issues/5#issuecomment-559478230

**Comment by durip** [comment]
everything is working lika a charm :)
OSMC Kodi 18.5

**Comment by ferdabasek** [comment]
- **fix works on KODI 18.5.**
- on KODI 19.0. - ALPHA1 plugin cannot run

 
[kodi.log](https://github.com/arvvoid/plugin.video.hbogoeu/files/3903046/kodi.log)

obrazek

**Comment by arvvoid** [comment]
@ferdabasek Tnx for testing. Yes on Kodi 19 Python3 Windows Build this is expected since pycryptodome is broken in that build (https://forum.kodi.tv/showthread.php?tid=348495, https://forum.kodi.tv/showthread.php?tid=348400).
For Kodi 19 Python3 testing of this add-on (or any add-on having cryptodome as a dependency) until this is fixed in the Kodi 19 Windows build only MacOS or Linux builds that use the system Python3 installation is possible (is just necessary to install pycryptodomex on the system using pip). It might work on the Android Kodi 19 test build but didn't try that one so idk if its reliable for testing.
A LibreElec test build with Kodi 19 Python3 exist as well I think.

But these changes should work fine on Python3.

**Comment by arvvoid** [comment]
I think this is ready for merging if any minor issue comes up at a later time can be easily fixed.

Since this prevents the add-on to be used by many users will speed things up merge and push a new release soon.

**Comment by arvvoid** [comment]
Fix included in latest release: https://github.com/arvvoid/plugin.video.hbogoeu/releases

**Pull request merged**